extras/rya.streams/query-manager/src/main/java/org/apache/rya/streams/querymanager/QueryManager.java

```diff
@@ -0,0 +1,241 @@
+            queryExecutor.startAndWait();
+        }
+
+        try {
+            queryExecutor.startQuery(ryaInstanceName, query);
+        } catch (final QueryExecutorException e) {
+            LOG.error("Failed to start query.", e);
+        }
+    }
+
+    /**
+     * Stops the specified query from running.
+     *
+     * @param queryId - The ID of the query to stop running. (not null)
+     */
+    private void stopQuery(final UUID queryId) {
+        requireNonNull(queryId);
+
+        LOG.trace("Stopping query: " + queryId.toString());
+        if (!queryExecutor.isRunning()) {
+            throw new IllegalStateException("Query Executor must be started before queries can be managed.");
+        }
+
+        try {
+            queryExecutor.stopQuery(queryId);
+        } catch (final QueryExecutorException e) {
+            LOG.error("Failed to stop query.", e);
+        }
+    }
+
+    @Override
+    protected void startUp() throws Exception {
+        LOG.trace("Starting Query Manager.");
+        queryRepos.forEach(repo -> repo.startAndWait());
+        sources.forEach((ryaInstance, source) -> source.startAndWait());
+    }
+
+    @Override
+    protected void shutDown() throws Exception {
+        LOG.trace("Stopping Query Manager.");
+        queryRepos.forEach(repo -> repo.stop());
+        sources.forEach((ryaInstance, source) -> source.stop());
+    }
+
+    /**
+     * An implementation of {@link QueryChangeLogListener} for the
+     * {@link QueryManager}.
+     * <p>
+     * When notified of a {@link ChangeType} performs one of the following:
+     * <li>{@link ChangeType#CREATE}: Creates a new query using the
+     * {@link QueryExecutor} provided to the {@link QueryManager}</li>
+     * <li>{@link ChangeType#DELETE}: Deletes a running query by stopping the
+     * {@link QueryExecutor} service of the queryID in the event</li>
+     * <li>{@link ChangeType#UPDATE}: If the query is running and the update is
+     * to stop the query, stops the query. Otherwise, if the query is not
+     * running, it is removed.</li>
+     */
+    private class QueryManagerQueryChange implements QueryChangeLogListener {
+        @Override
+        public void notify(final ChangeLogEntry<QueryChange> queryChangeEvent) {
+            LOG.debug("New query change event.");
+            final QueryChange entry = queryChangeEvent.getEntry();
+
+            switch (entry.getChangeType()) {
+                case CREATE:
+                    LOG.debug("Creating query event.");
+                    if (entry.getIsActive().or(false)) {
+                        final StreamsQuery newQuery = new StreamsQuery(entry.getQueryId(), entry.getSparql().get(),
+                                entry.getIsActive().or(false));
+                        runQuery(newQuery);
+                        queriesCache.put(entry.getQueryId(), newQuery);
+                        LOG.trace("Created new query: " + newQuery.toString());
```

Comment:
I'm not sure how you're choosing to log at debug and trace, but it seems like all of this should be debug. We also should print an info whenever we're changing query execution stuff in general, otherwise it's going to be hard to tell that this process is doing anything when deployed.

Reply:
yeah got trace/info mixed up.